src/recipe_dictionary.h

```diff
@@ -133,6 +133,10 @@ class recipe_subset
         /** Find recipes matching query (left anchored partial matches are supported) */
         std::vector<const recipe *> search( const std::string &txt,
                                             const search_type key = search_type::name ) const;
+        /** Find recipes matching query existing in subset of recipes (left anchored partial matches are supported) */
+        std::vector<const recipe *> search( const std::vector<const recipe *> subset,
```

Comment:
It's a bit strange to be asking a `recipe_subset` object to filter a different subset of recipes.  The name suggests that the `recipe_subset` already represents a subset of recipes.  So this interface is confusing (though I don't blame you for taking this approach; it's straightforward).

I feel like a more logical solution (which requires more changes) is that `recipe_subset::search` should not return a `vector<recipe*>`, but instead return a new `recipe_subset` object representing the new, smaller subset.  Then you would not need to add this new function; it would simply be a matter of repeated `search` calls.  That would naturally fix some of my objections above.

This is just an idea; I haven't thought it through in detail; it may not work out in practice.

Reply:
Works surprisingly well actually! Gets rid of that second `search` function, made a new function `reduce` that calls into `search` before returning a `recipe_subset` reducible by future `reduce` calls :D